### DISCUSSION TITLE
What are the possible network connection types?

### DISCUSSION THREAD

**Discussion by coffeymatt** [ionic-team/capacitor#2444] [Q&A]
I want to detect whether the user has a data connection (not just cellular/voice), but it's not clear how to do that from the docs. I thought this could be determined from the connection type, but the potential values for this property are not well defined or documented. 

When I look at the interface for what 'Network.getStatus()' returns it says:

connectionType: 'wifi' | 'cellular' | 'none' | 'unknown';

But when I test in a PWA it returns none of these values (in fact it returns '4g' despite being on ethernet). Can the docs and this interface be updated with the potential values this can be, and perhaps provide a simple boolean property for 'data' rather than just 'connected' (as that could just be voice right?) ?

**Comment by dotjiwa** [comment]
Im experiencing the exact same. Always getting 4g while connected to wifi.

**Comment by jcesarmobile** [comment]
For Capacitor 3 we are moving plugins out of Capacitor.
The new @capacitor/network plugin handles this properly
https://github.com/ionic-team/capacitor-plugins/tree/main/network

**Reply by dotjiwa** [reply]
Thanks @jcesarmobile .

I've tried installing this version using 'npm install @capacitor/network'.. which did not work. 

Whats the preferred way of installing this version?

**Reply by jcesarmobile** [reply]
It's not released yet and will require to use capacitor 3

**Reply by dotjiwa** [reply]
Ah I see! Any ETA? :)
